x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/components/fleet_extensions/aws_credentials_form/aws_credentials_form.test.tsx

```diff
@@ -0,0 +1,351 @@
+    isValid?: boolean;
+  }
+
+  const renderWithProviders = (props: AwsCredentialsFormTestProps) => {
+    return render(
+      <I18nProvider>
+        <AwsCredentialsForm
+          {...props}
+          disabled={props.disabled ?? false}
+          hasInvalidRequiredVars={props.hasInvalidRequiredVars ?? false}
+          isValid={props.isValid ?? true}
+        />
+      </I18nProvider>
+    );
+  };
+
+  beforeEach(() => {
+    mockUpdatePolicy.mockClear();
+    mockAwsCredentialTypeSelector.mockClear();
+    mockAwsInputVarFields.mockClear();
+    mockAWSSetupInfoContent.mockClear();
+    mockUseCloudSetup.mockReturnValue(
+      createAwsCloudSetupMock({
+        shortName: 'CSPM',
+        templateName: 'cspm',
+      })
+    );
+  });
+
+  describe('Component Rendering', () => {
+    it('renders default', () => {
+      renderWithProviders(defaultProps);
+
+      // Verify AWSSetupInfoContent is rendered
+      expect(screen.getByTestId('aws-setup-info')).toBeInTheDocument();
+      expect(screen.getByText('Getting Started')).toBeInTheDocument();
+
+      // Verify RadioGroup renders both setup format options
+      const cloudFormationRadio = screen.getByTestId(
+        AWS_CREDENTIALS_TYPE_OPTIONS_TEST_SUBJECTS.CLOUDFORMATION
+      );
+      const manualRadio = screen.getByTestId(AWS_CREDENTIALS_TYPE_OPTIONS_TEST_SUBJECTS.MANUAL);
+
+      expect(cloudFormationRadio).toBeInTheDocument();
+      expect(manualRadio).toBeInTheDocument();
+      expect(screen.getByRole('button', { name: 'CloudFormation' })).toBeInTheDocument();
+      expect(screen.getByRole('button', { name: 'Manual' })).toBeInTheDocument();
+    });
+
+    it('disables setup format options when disabled prop is true', () => {
+      renderWithProviders({ ...defaultProps, disabled: true });
+
+      // Verify both radio buttons are disabled
+      expect(screen.getByRole('button', { name: 'CloudFormation' })).toBeDisabled();
+      expect(screen.getByRole('button', { name: 'Manual' })).toBeDisabled();
+    });
+  });
+
+  describe('CloudFormation Setup', () => {
+    it('renders CloudFormation radio as selected when CloudFormation format is selected', () => {
+      const propsWithCloudFormation = {
+        ...defaultProps,
+        input: {
+          ...defaultProps.input,
+          streams: [
+            {
+              enabled: true,
+              data_stream: defaultProps.input.streams[0].data_stream,
+              vars: {
+                'aws.credentials.type': { value: 'cloud_formation' },
+              },
+            },
+          ],
+        },
+      };
+
+      renderWithProviders(propsWithCloudFormation);
+
+      // Verify both setup format buttons are present
+      expect(screen.getByRole('button', { name: 'CloudFormation' })).toBeInTheDocument();
+      expect(screen.getByRole('button', { name: 'Manual' })).toBeInTheDocument();
+    });
+
+    it('shows warning when CloudFormation template is not supported', () => {
```

Comment:
I couldn't understand what makes this test a specific test to test the warning. The only difference  between `propsWithCloudFormation` and `propsWithNoTemplate` in the `packageInfo.name` , but that's not what makes it "not support cloudformation" according to the code. In the code I see that we check for `cloud_formation_template` in the vars. So I assuming if I make  this assertion in the first test, it will also pass
```
expect(
        screen.getByText(/CloudFormation is not supported on the current Integration version/)
      ).toBeInTheDocument();
```